dialogue-core/src/main/java/com/palantir/dialogue/core/NodeSelectionStrategyChannel.java

```diff
@@ -74,11 +75,11 @@
 
     static LimitedChannel create(Config cf, ImmutableList<LimitedChannel> channels) {
         if (channels.isEmpty()) {
-            return new ZeroUriNodeSelectionChannel(cf.channelName());
+            return new StickyChannelHandler(new ZeroUriNodeSelectionChannel(cf.channelName()));
```

Comment:
```suggestion
            // If we reloaded away from a channel, we should still attempt to allow remaining requests
            // to execute against the old reference.
            return new StickyChannelHandler(new ZeroUriNodeSelectionChannel(cf.channelName()));
```

Reply:
+1

Reply:
I don't think this is a concern: reload creates a new DialogueChannel; existing sticky channels (as defined here)https://github.com/palantir/dialogue/pull/1334/files will continue referring to the old DialogueChannel instance; by construction (as long as we make StickyAttachments package private) there is should be no way for requests from one DialogueChannel (and therefore one set of LimitedChannels) to bleed into another DialogueChannel.